src/utils/Logging.ts

```diff
@@ -658,36 +658,42 @@ export default class Logging {
   }
 
   private static anonymizeSensitiveData(message: any) {
-    if (!message) {
+    if (!message || typeof message === 'number' || typeof message === 'boolean' || typeof message === 'function') {
+      // eslint-disable-next-line no-useless-return
       return;
-    }
-    if (typeof message === 'object') {
-      for (const key in message) {
-        if (message[key]) {
-          const value = message[key];
-          // Another JSon?
-          if (typeof value === 'object') {
-            Logging.anonymizeSensitiveData(message[key]);
-          }
-          // Array?
-          if (Array.isArray(value)) {
-            Logging.anonymizeSensitiveData(value);
-          }
-          // String?
-          if (typeof value === 'string') {
-            if (key === 'password' ||
-                key === 'repeatPassword' ||
-                key === 'captcha') {
-              // Anonymize
-              message[key] = Constants.ANONYMIZED_VALUE;
-            }
-          }
-        }
-      }
+    } else if (typeof message === 'string') {
+      // Anonymize
+      message.replace(/((repeat|)[pP]assword|captcha)(\s)(=|:)(\s)(.*)/g, '$1$3$4$5' + Constants.ANONYMIZED_VALUE);
```

Comment:
Not understandable by a human being.

Reply:
The regex is simple, more simple than the ones you can read in Constants.ts.
The replace https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/String/replace documentation is clear enough to explain what is done.